src/api/users.coffee

```diff
@@ -42,16 +43,91 @@ exports.authenticate = (email) ->
     utils.logAndSetResponse this, 500, "Error during authentication #{e}", 'error'
 
 
+
+#################################
+### Reset password Functions ###
+#################################
+
+
+passwordResetPlainMessageTemplate = (firstname, setPasswordLink) -> """
+<---------- Existing User - Reset Password ---------->
+Hi #{firstname},
+
+A request has been made to reset your password on OpenHIM
+Follow the below link to reset your password and log into OpenHIM Console
+#{setPasswordLink}
+<---------- Existing User - Reset Password ---------->
+"""
+
+passwordResetHtmlMessageTemplate = (firstname, setPasswordLink) -> """
+<h1>Reset OpenHIM Password</h1>
+<p>Hi #{firstname},<br/><br/>A request has been made to reset your password on OpenHIM</p>
+<p>Follow the below link to set your password and log into OpenHIM Console</p>
+<p>#{setPasswordLink}</p>
+"""
+
+###
+# update user token/expiry and send new password email
+###
+exports.userPasswordResetRequest = (email) ->
+  email = unescape email
+
+  if email is 'root@openhim.org'
+    this.body = "Cannot request password reset for 'root@openhim.org'"
+    this.status = 403
+    return
+
+  # Generate the new user token here
+  # set expiry date = true
+
+  token = randtoken.generate 32
+  duration = config.userPasswordResetExpiry.duration
+  durationType = config.userPasswordResetExpiry.durationType
+  expiry = moment().add(duration, durationType).utc().format()
+
+  updateUserTokenExpiry =
+    token: token
+    tokenType: 'existingUser'
+    expiry: expiry
+
+  try
+    user = yield User.findOneAndUpdate(email: email, updateUserTokenExpiry).exec()
+
+    if not user
+      this.body = "Tried to request password reset for invalid email address: #{email}"
+      this.status = 404
+      logger.info "Tried to request password reset for invalid email address: #{email}"
+      return
+
+    consoleURL = config.alerts.consoleURL
+    setPasswordLink = "#{consoleURL}/#/set-password/#{token}"
+    
+    # Send email to user to reset password
+    plainMessage = passwordResetPlainMessageTemplate user.firstname, setPasswordLink
+    htmlMessage = passwordResetHtmlMessageTemplate user.firstname, setPasswordLink
+
+    contact.contactUser 'email', email, 'OpenHIM Console Password Reset', plainMessage, htmlMessage, ->
```

Comment:
`contactUser` has an error parameter in the callback. You should check for this and log and respond with a 500 status if present.